test/framework/shootmigrationtest.go

```diff
@@ -0,0 +1,286 @@
+)
+
+const (
+	SecretName              = "test-shoot-migration-secret"
+	SecretNamespace         = "default"
+	ServiceAccountName      = "test-service-account"
+	ServiceAccountNamespace = "default"
+)
+
+// ShootMigrationTest represents the seed test framework that includes
+// test functions that can be executed ona specific seed
+type ShootMigrationTest struct {
+	GardenerFramework                 *GardenerFramework
+	Config                            *ShootMigrationConfig
+	TargetSeedClient                  kubernetes.Interface
+	SourceSeedClient                  kubernetes.Interface
+	ShootClient                       kubernetes.Interface
+	TargetSeed                        *gardencorev1beta1.Seed
+	SourceSeed                        *gardencorev1beta1.Seed
+	ComparisonElementsBeforeMigration ShootComparisonElements
+	ComparisonElementsAfterMigration  ShootComparisonElements
+	Shoot                             gardencorev1beta1.Shoot
+	SeedShootNamespace                string
+}
+
+// ShootMigrationConfig is the configuration for a seed framework that will be filled with user provided data
+type ShootMigrationConfig struct {
+	TargetSeedName string
+	SourceSeedName string
+	ShootName      string
+	ShootNamespace string
+}
+
+// ShootComparisonElements contains details about Machines, Nodes and Pods that will be compared during the tests
+type ShootComparisonElements struct {
+	MachineNames []string
+	MachineNodes []string
+	NodeNames    []string
+	PodStatus    map[string]corev1.PodPhase
+}
+
+// NewShootMigrationTest creates a new simple shoot migration test
+func NewShootMigrationTest(f *GardenerFramework, cfg *ShootMigrationConfig) *ShootMigrationTest {
+	t := &ShootMigrationTest{
+		GardenerFramework: f,
+		Config:            cfg,
+	}
+	return t
+}
+
+// MigrateShoot triggers shoot migration by changing the value of "shoot.Spec.SeedName" to the value of "ShootMigrationConfig.TargetSeedName"
+func (t *ShootMigrationTest) MigrateShoot(ctx context.Context) error {
+	// Dump gardener state if delete shoot is in exit handler
+	if os.Getenv("TM_PHASE") == "Exit" {
+		if shootFramework, err := t.GardenerFramework.NewShootFramework(&t.Shoot); err == nil {
+			shootFramework.DumpState(ctx)
+		} else {
+			t.GardenerFramework.DumpState(ctx)
+		}
+	}
+
+	err := t.GardenerFramework.UpdateShoot(ctx, &t.Shoot, func(shoot *gardencorev1beta1.Shoot) error {
+		t := gardencorev1beta1.Toleration{}
+		t.Key = core.SeedTaintProtected
+
+		if shoot.Spec.Tolerations == nil {
+			shoot.Spec.Tolerations = make([]gardencorev1beta1.Toleration, 0)
+		} else {
+			for _, t := range shoot.Spec.Tolerations {
+				if t.Key == core.SeedTaintProtected {
+					return nil
+				}
+			}
+		}
+		shoot.Spec.Tolerations = append(shoot.Spec.Tolerations, t)
+
+		return nil
+	})
+	if err != nil {
+		return err
+	}
+
+	return t.GardenerFramework.MigrateShoot(ctx, &t.Shoot, t.TargetSeed)
+}
+
+// GetNodeNames is using the seedClient to fetch all Node names from the Seed
```

Comment:
```suggestion
// GetNodeNames uses the shootClient to fetch all Node names from the Shoot
```
We should be getting the nodes of the shoot. I see that when the method is called the shootClient is actually passed, so can you change the names of the variables.